pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -0,0 +1,244 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package model
+
+import (
+	"errors"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/pointer"
+	datadoghq "github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// UnsetGeneration is the value used to represent a .Spec value for which we have no generation (not created or not updated in-cluster yet)
+const UnsetGeneration = -1
+
+// PodAutoscalerInternal hols the necessary data to work with the `DatadogPodAutoscaler` CRD.
+type PodAutoscalerInternal struct {
+	// Namespace is the namespace of the PodAutoscaler
+	Namespace string
+
+	// Name is the name of the PodAutoscaler
+	Name string
+
+	// Generation is the received generation of the PodAutoscaler
+	Generation int64
+
+	// Keeping track of .Spec (configuration of the Autoscaling)
+	Spec datadoghq.DatadogPodAutoscalerSpec
+
+	// TargetGVK is the GroupVersionKind of the target resource
+	// Parsed once from the .Spec.TargetRef
+	TargetGVK schema.GroupVersionKind
+
+	// SettingsVersion is the version of the settings from RC
+	// (only if owner == remote)
+	SettingsVersion *uint64
+
+	// SettingsTimestamp is the time when the settings were last updated
+	// (only if owner == remote)
+	SettingsTimestamp time.Time
+
+	// Target represents the desired state
+	ScalingValues ScalingValues
```

Comment:
IMO renaming this field to `DesiredState` would make the code much easier to read